formats/src/main/scala/overflowdb/formats/neo4jcsv/Neo4jCsvExporter.scala

```diff
@@ -0,0 +1,92 @@
+package overflowdb.formats.neo4jcsv
+
+import com.github.tototoshi.csv._
+import overflowdb.Graph
+import overflowdb.formats.Exporter
+import overflowdb.traversal.Traversal
+
+import java.nio.file.Path
+import scala.collection.mutable
+import scala.jdk.CollectionConverters.{CollectionHasAsScala, MapHasAsScala}
+import scala.jdk.OptionConverters.RichOptional
+import scala.util.Using
+
+object Neo4jCsvExporter extends Exporter {
+
+  /**
+   * Exports OverflowDB Graph to neo4j csv files
+   * see https://neo4j.com/docs/operations-manual/current/tools/neo4j-admin/neo4j-admin-import/
+   *
+   * For both nodes and relationships, we first write the data file and to derive the property types from their
+   * runtime types. We will write columns for all declared properties, because we only know which ones are
+   * actually in use *after* traversing all elements.
+   * */
+  override def runExport(graph: Graph, outputRootDirectory: Path): Seq[Path] = {
+    val labelsWithNodes = graph.nodeCountByLabel.asScala.collect {
+      case (label, count) if count > 0 => label
+    }.toSeq
+
+    val nodeFiles = labelsWithNodes.flatMap(exportNodes(graph, _, outputRootDirectory))
+    val edgeFiles = exportEdges(graph, outputRootDirectory)
+    nodeFiles ++ edgeFiles
+  }
+
+  private def exportNodes(graph: Graph, label: String, outputRootDirectory: Path): Seq[Path] = {
+    val dataFile   = outputRootDirectory.resolve(s"$label.csv")
+    val headerFile = outputRootDirectory.resolve(s"$label$HeaderFileSuffix.csv")  // to be written at the very end, with complete ColumnDefByName
+    val columnDefinitions = new ColumnDefinitions(graph.nodes(label).next.propertyKeys.asScala)
+
+    Using(CSVWriter.open(dataFile.toFile, append = false)) { writer =>
+      Traversal(graph.nodes(label)).foreach { node =>
```

Comment:
Any reason to prefer `Traversal(...).foreach` to:

```
graph.nodes(label).forEachRemaining { node =>
        val specialColumns = Seq(node.id.toString, node.label)
        val propertyValueColumns = columnDefinitions.propertyValues(node.propertyOption(_).toScala)
        writer.writeRow(specialColumns ++ propertyValueColumns)
      }
```

Reply:
not really, and yes you're right, using the simplest tool for the job would be the right thing to do. done